sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -18,112 +18,127 @@ object Klaviyo {
-     * UserInfo is saved to keep track of current profile details and
-     * used to autocomplete analytics requests with profile identifier when not specified
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
      *
      * This should be called whenever the active user in your app changes
      * (e.g. after a fresh login)
      *
      * @param email Email address for active user
+     * @return
      */
-    fun setEmail(email: String) = apply {
+    fun setEmail(email: String): Klaviyo = apply {
+        // TODO validate email format?
         UserInfo.email = email
+        this.setProfile(UserInfo.getAsCustomerProperties())
     }
 
     /**
-     * Assigns a phone number to the current UserInfo
+     * Assigns a phone number to the current internally tracked profile
      *
-     * UserInfo is saved to keep track of current profile details and
-     * used to autocomplete analytics requests with profile identifier when not specified
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
      *
      * This should be called whenever the active user in your app changes
      * (e.g. after a fresh login)
      *
      * @param phone Phone number for active user
      */
-    fun setPhone(phone: String) = apply {
-        UserInfo.phone = phone
+    fun setPhoneNumber(phone: String): Klaviyo = apply {
+        // TODO validate phone format?
+        UserInfo.phoneNumber = phone
+        this.setProfile(UserInfo.getAsCustomerProperties())
     }
 
     /**
-     * Clears all stored UserInfo identifiers (e.g. email or phone)
+     * Assigns an external to the current internally tracked profile
      *
-     * This should be called whenever an active user in your app is removed
-     * (e.g. after a logout)
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param id Phone number for active user
+     * @return
      */
-    fun reset() = apply {
-        UserInfo.reset()
+    fun setExternalId(id: String): Klaviyo = apply {
+        UserInfo.externalId = id
+        this.setProfile(UserInfo.getAsCustomerProperties())
     }
 
-    //endregion
-
-    //region Analytics API
-
     /**
      * Queues a request to identify profile properties to the Klaviyo API
      * Identify requests track specific properties about a user without triggering an event
      *
      * @param properties A map of properties that define the user
+     * @return
      */
-    fun createProfile(properties: KlaviyoCustomerProperties) {
+    fun setProfile(properties: KlaviyoCustomerProperties): Klaviyo = apply {
+        // TODO Extract phone/email and save in state?
+        //  I'm still a bit confused about the exact purpose of this method
```

Comment:
This was originally just a call-through to issue an identity request to Klaviyo. I know it got renamed to match the iOS SDK a bit closer, but is there anything in particular you are confused about here?

Reply:
Oh yeah thanks for highlighting that. 
What is unclear to me is how the profile properties passed in as the argument should relate to the properties in internal state. E.g. if email is set in this argument's properties, should that email replace the current email in state? If email is _not_ specified, should we add the current email from state? Should it be a pure pass through that doesn't touch internal state or read from it?

Reply:
Oo I see, good point. Since we weren't really dealing with state before, we just exposed the function with a map of customer properties as an input but if we start doing proper persistence in the SDK this makes less sense.

It could be confusing for internal state to overwrite what a customer may pass in here (and they likely will want to pass in some information here since this takes in properties beyond the basic identifiers we store). I wonder if we should take the route where we overwrite whats stored in our persistence layer with any identifier values that have been attached to the customer properties map.

I could see this going multiple ways based on different opinions

Reply:
Yeah, and I think it depends on naming too. `createProfile` sounds a little more like a straight up pass-through that ignores state. `setProfile` sounds more like its going to overwrite internal state (and subsequently make the API call). 
And by the way thinking about this is what prompted me to propose a `setProfileAttribute` method where one could update any properties without having to specify the a whole profile object. Just realizing now I didn't put that method in this PR, but it was in the google doc.

Reply:
I'll push another commit with a proposal for merging this incoming argument with the internal state

Reply:
@ndurell your input would be valuable too. I _think_ what Jordan and I are talking about is similar to how `trackPersonWithInfo` works, which would be the iOS analog until you rename things.

Reply:
Updated, [see here](https://github.com/klaviyo/klaviyo-android-sdk/pull/31/files#r1066335425) added a two-way merge for this situation.